### PR TITLE
[SELC-4892] Added function to send events on topic through event hub

### PR DESCRIPTION
**Pull request by pierpaolodidato89**
#### List of Changes

- Added trigger endpoint to receive onboarding updates
- Added function to send events on topic through event hub
- Added event hub rest client and its configuration to send messages for different products

#### Motivation and Context

In order to manage all events from database, a new trigger endpoint has been been added. This endpoint invoke an azure function to send messages on topic

#### How Has This Been Tested?

I have deployed the feature branch direclty in DEV environment.

### CODE REVIEW

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java [hunk 1]
```diff
@@ -139,6 +146,60 @@ public void onboardingsOrchestrator(
...
+        if (request.getBody().isEmpty()) {
+            return request.createResponseBuilder(HttpStatus.BAD_REQUEST)
+                    .body("\"Request body cannot be empty.")
+                    .build();
+        }
+
+        final String onboarding = request.getBody().get();
+        final String timeoutString = request.getQueryParameters().get("timeout");
+        DurableTaskClient client = durableContext.getClient();
+        String instanceId = client.scheduleNewOrchestrationInstance(SEND_ONBOARDING_NOTIFICATION, onboarding);
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java:172]: you could avoid to use durable function only for calling one function....you could just use an http azure function

Comment (gunzip) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java:172]: agree

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java [hunk 2]
```diff
@@ -36,14 +37,20 @@ public class OnboardingFunctions {
 
     private final OnboardingService service;
     private final CompletionService completionService;
+    private final MessageService messageService;
 
     private final ObjectMapper objectMapper;
     private final TaskOptions optionsRetry;
 
-    public OnboardingFunctions(OnboardingService service, ObjectMapper objectMapper, RetryPolicyConfig retryPolicyConfig, CompletionService completionService) {
+    public OnboardingFunctions(OnboardingService service,
```
Comment (manuraf) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java:45]: you could create another class for this specific use case, for ex. NotificationFunctions

**Comment by manuraf** [comment]
You could use a different approach instead iterator on product.consumers configuration....you might create 2 independent functions that listened to the event on SC-Contract and then sent to SC-ContractFD and SC-ContractSAP with the related mapping logic

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java (hunk 2 above)
Comment (pierpaolodidato89) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctions.java:42]: As you suggested, I created a separated class to host logic for notifications functions

**Review by andrea-putzu — APPROVED** [review]

**Pull request merged**